pwiz/data/vendor_readers/Bruker/ChromatographyDataSqliteReader.cpp

```diff
@@ -0,0 +1,366 @@
+    Viscosity_cP = 36,
+    Voltage_kV = 37,
+    Voltage_V = 38,
+    Voltage_mV = 39,
+    Volume_l = 40,
+    Volume_ml = 41,
+    Volume_mul = 42,
+    Energy_J = 43,
+    Energy_mJ = 44,
+    Energy_muJ = 45,
+    Length_Angstrom = 46,
+};
+
+
+using pwiz::vendor_api::Bruker::Chromatogram;
+using pwiz::vendor_api::Bruker::ChromatogramPtr;
+
+// Forward declarations for local helpers used below
+static CVID traceTypeToCVID(int type, int unit, const std::string description);
+static CVID traceUnitToCVID(int unit, double& value);
+static TraceUnit getDefaultUnitForTraceType(CVID chromatogramType);
+
+std::vector<vendor_api::Bruker::ChromatogramPtr> readChromatographyDataSqlite(const std::string rootpath)
+{
+    std::vector<vendor_api::Bruker::ChromatogramPtr> results;
+    try
+    {
+        bfs::path chromFilepath = bfs::path(rootpath) / "chromatography-data.sqlite";
+        if (!bfs::exists(chromFilepath))
+            return results;
+
+        sqlite3pp::database db(chromFilepath.string());
+        sqlite3pp::query q(db, "SELECT Id FROM TraceSources ORDER BY Id");
+
+        for (sqlite3pp::query::iterator itr = q.begin(); itr != q.end(); ++itr)
+        {
+            int traceId = itr->get<int>(0);
+
+            ChromatogramPtr chrom(new Chromatogram);
+
+            int unitValue = 0;
+            double timeOffset = 0.0;
+
+            // Get trace metadata from TraceSources
+            std::ostringstream sourceQuery;
+            sourceQuery << "SELECT Type, Unit, Description, Instrument, TimeOffset FROM TraceSources WHERE Id=" << traceId;
+            sqlite3pp::query sq(db, sourceQuery.str().c_str());
+            sqlite3pp::query::iterator sitr = sq.begin();
+            if (sitr != sq.end())
+            {
+                chrom->description = sitr->get<std::string>(2);
+                chrom->instrument = sitr->get<std::string>(3);
+                timeOffset = sitr->get<double>(4);
+                unitValue = sitr->get<int>(1);
+                chrom->type = traceTypeToCVID(sitr->get<int>(0), unitValue, chrom->description);
+                if (unitValue == static_cast<int>(TraceUnit::NoneUnit))
+                {
+                    if (chrom->type == CVID::MS_pressure_chromatogram && bal::contains(chrom->description, "[psi]"))
+                        unitValue = static_cast<int>(TraceUnit::Pressure_psi);
+                    else
+                        unitValue = static_cast<int>(TraceUnit::Intensity); // default to Intensity if no unit specified
+                }
+                double dummyValue = 0.0;
+                chrom->units = traceUnitToCVID(unitValue, dummyValue);
+            }
+
+            sqlite3_stmt* chunksStmt = nullptr;
+            std::ostringstream chunksQuery;
+            chunksQuery << "SELECT Times, Intensities FROM TraceChunks WHERE Trace=? ORDER BY ROWID";
+
+            if (sqlite3_prepare_v2(db.connected(), chunksQuery.str().c_str(), -1, &chunksStmt, nullptr) == SQLITE_OK)
```

Comment:
Why use raw sqlite API here?

Reply:
Open to other approaches - just modeling on the python that Sven provided